### PR TITLE
TYP #37715: Fix mypy errors in column.py

### PR DESCRIPTION
**Pull request by bang128**
- [ ] xref #37715 
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/core/interchange/column.py [hunk 1]
```diff
@@ -81,7 +81,7 @@ def __init__(self, column: pd.Series, allow_copy: bool = True) -> None:
         self._col = column
         self._allow_copy = allow_copy
 
-    def size(self) -> int:  # type: ignore[override]
+    def size(self) -> int:
```
Comment (WillAyd) [inline pandas/core/interchange/column.py:84]: I think you need to add the `@property` decorator to get this to work with mypy

**Review by WillAyd — CHANGES_REQUESTED** [review]

File: pandas/core/interchange/column.py (hunk 1 above)
Comment (bang128) [reply pandas/core/interchange/column.py:84]: I added it in my first commit, but the tests failed, so I tried removing @property

**Comment by WillAyd** [comment]
Huh strange - I guess that's why this was ignored before. That said, I think the property data is correct, at least according to the dataframe protocol

https://data-apis.org/dataframe-protocol/latest/API.html

So I think you can update the tests and add a whatsnew note describing this as a bug fix

@jorisvandenbossche

**Comment by jorisvandenbossche** [comment]
> That said, I think the property decorator is correct, at least according to the dataframe protocol
> 
> https://data-apis.org/dataframe-protocol/latest/API.html

Hmm, the documentation seems to be not up to date. This was originally a property, but changed to a method a while ago (https://github.com/data-apis/dataframe-api/pull/74#discussion_r841112693), so the currently implementation here is actually correct.

**Comment by mroeschke** [comment]
Thanks for the pull request, but I believe this was addressed and fixed in #50565 so closing. Happy to have your contributions on any other typing issues!

**Pull request closed without merging**